src/completers/git.zsh

```diff
@@ -720,6 +720,33 @@ _fzf_complete_git-commit-messages_post() {
     echo ${(qq)message}
 }
 
+_fzf_complete_git-ls-files() {
+    local git_options=$1
+    local fzf_options=$2
+    shift 2
+
+    _fzf_complete --ansi --read0 --print0 ${(Q)${(Z+n+)fzf_options}} -- $@ < <({
+        local cdup=$(git rev-parse --show-cdup 2> /dev/null)
+        local git_prefix=$(git rev-parse --show-prefix 2> /dev/null)
+        cd $(git rev-parse --show-toplevel 2> /dev/null)
+
+        local files=$(git ls-files --deduplicate -z ${(Z+n+)git_options} 2> /dev/null)
+        files=(${(0)files})
+        local paths=($cdup${^files})
```

Comment:
A quick note for the possible later reference:
```
${^spec}
        Turn  on  the RC_EXPAND_PARAM option for the evaluation of spec; if the `^' is doubled, turn it off.  When this option is set,
        array expansions of the form foo${xx}bar, where the parameter xx is set to (a b c),  are  substituted  with  `fooabar  foobbar
        foocbar' instead of the default `fooa b cbar'.  Note that an empty array will therefore cause all arguments to be removed.

        Internally,  each  such  expansion  is  converted  into  the  equivalent  list  for  brace  expansion.   E.g., ${^var} becomes
        {$var[1],$var[2],...}, and is processed as described in the section `Brace Expansion' below: note, however, the expansion hap‐
        pens  immediately,  with  any  explicit  brace expansion happening later.  If word splitting is also in effect the $var[N] may
        themselves be split into different list elements.
```

Reply:
This expression is pretty brilliant ✨